daemon/cdi_linux.go

```diff
@@ -0,0 +1,63 @@
+package daemon
+
+import (
+	"fmt"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	"github.com/docker/docker/pkg/capabilities"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+// RegisterCDIDriver registers the CDI device driver.
+// The driver injects CDI devices into an incoming OCI spec and is called for DeviceRequests associated with CDI devices.
+func RegisterCDIDriver(opts ...CDIOption) {
+	c := &cdiHandler{}
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	if len(c.specDirs) == 0 {
+		c.specDirs = cdi.DefaultSpecDirs
+	}
+
+	driver := &deviceDriver{
+		capset:     capabilities.Set{"cdi": struct{}{}},
+		updateSpec: c.injectCDIDevices,
+	}
+
+	registerDeviceDriver("cdi", driver)
+}
+
+// injectCDIDevices injects a set of CDI devices into the specified OCI specification.
+// Note that although the CDI specification can be applied to devices on platforms other than linux, the package used
+// currently only supports CDI device injection. If this restriction were to be relaxed, the function below should be usable as-is.
+func (c *cdiHandler) injectCDIDevices(s *specs.Spec, dev *deviceInstance) error {
+	if dev == nil {
+		return nil
+	}
+
+	cdiDeviceNames := dev.req.DeviceIDs
+	if len(cdiDeviceNames) == 0 {
+		return nil
+	}
+
+	registry := cdi.GetRegistry(
```

Comment:
I just caught onto what `cdi.GetRegistry` does. I will not approve a PR which contains any references to that function.

Reply:
I have removed the use of `cdi.GetRegistry` and now instantiate a `cdi.Cache` by calling `cdi.NewCache` directly.